### PR TITLE
fix #274870: fix resetting shortcuts

### PR DESCRIPTION
**Pull request by dmitrio95**
This pull request is primarily to fix [this issue](https://musescore.org/en/node/274870). The problem was in reinitializing shortcuts at each program start when setting a shortcut scheme based on user's keyboard layout. The first commit of this PR makes it happen only if "shortcuts.xml" file is not found and thus default shortcuts should be used. The second commit makes preferences dialog be more reliable in resetting keyboard shortcuts to default values in some cases — I didn't find any corresponding bug report but I can file one in case it is needed.

These commits can be squashed before merge, I left them separate to make reviewing them simpler.

### CODE REVIEW

File: mscore/musescore.cpp [hunk 1]
```diff
@@ -6810,9 +6810,12 @@ int main(int argc, char* av[])
                   preferences.setToDefaultValue(PREF_APP_PATHS_MYSOUNDFONTS);
                   preferences.setToDefaultValue(PREF_APP_PATHS_MYEXTENSIONS);
                   updateExternalValuesFromPreferences();
+
```
[the remark this answers is not in the document]
Comment (dmitrio95) [reply mscore/musescore.cpp]: Thanks, fixed that

File: mscore/shortcut.h [hunk 2]
```diff
@@ -115,12 +115,16 @@ class Shortcut {
...
 
+      static QString source;
+
       static Shortcut _sc[];
       static QHash<QByteArray, Shortcut*> _shortcuts;
       void translateAction(QAction* action) const;
 
    public:
 
+      static constexpr const char* defaultLocation = ":/data/shortcuts.xml";
```
Comment (anatoly-os) [inline mscore/shortcut.h]: Filename is not a location. Let's find better name like "Default file name".

**Comment by anatoly-os** [comment]
Do make, please.

**Comment by dmitrio95** [comment]
Ah, I forgot that this only happens after applying the first commit, otherwise all shortcuts are reset to default values anyway.
But if you apply it, the following should reproduce it:
1) In preferences reassign shortcut for "File: Save" action to Ctrl+Shift+S.
2) Relaunch MuseScore.
3) Go to preferences and reset a shortcut to "File: Save" to default.
After closing preferences file save action will have no shortcut assigned, and both Ctrl+S and Ctrl+Shift+S will do nothing.
This happens since the default value for file save action is set via Qt standard keys, and this was not correctly handled by the preferences dialog.

**Comment by SchnennartLeider** [comment]
Hey everyone, I am totally new here, and I just want to fix that bug, that I cant save or load any shortkeys in musescore 3 - but how do I do it? 
Do I have to manually add the 'commits' in the respective files? Where exactly, if so?
Thanks!

**Comment by dmitrio95** [comment]
@SchnennartLeider Hello! We'll be glad to help you but a pull request on Github is not a proper place to discuss such questions. You may find some useful development info in [the developer's handbook](https://musescore.org/en/handbook/developers-handbook), in case you have some questions feel free to ask on [the forum](https://musescore.org/en/forum) or join [our Telegram chat](https://t.me/musescoreeditorchat).

**Pull request merged**